### PR TITLE
[Twitter Analytics]  Use WidgetChannelManager from base widget 

### PR DESCRIPTION
**Pull request by GowthamyVaseekaran**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes
.

### CODE REVIEW

File: modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx [hunk 1]
```diff
@@ -55,21 +54,22 @@ class Emotions extends Widget {
...
-            type: 'RDBMSBatchDataProvider',
-            config: {
-                datasourceName: 'Twitter_Analytics',
-                queryData:{
-                    query: "select type as Sentiment, count(TweetID) as Rate from sentiment where PARSEDATETIME(timestamp, 'yyyy-mm-dd hh:mm:ss','en') > CURRENT_TIMESTAMP()-86400 group by type"
-                },
-                tableName: 'sentiment',
-                incrementalColumn: 'Sentiment',
-                publishingInterval: 60,
+            configs: {
```
Comment (ramindu90) [inline modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx]: @GowthamyVaseekaran : we can move this to widgetconf.json file. So the config can be configurable for the user. Please check http-analytics or activity-explorer gadgets

Comment (Niveathika92) [reply modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx]: We were discussing updating the base widget to have this functionality(i.e get data provider conf in one function call). Shall we update this after that change is integrated to the base widget?

Comment (Niveathika92) [reply modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx]: @GowthamyVaseekaran Wso2-dashboard 1.2.1 is released, data provider confs can be moved to the widgetconf file easily now.

Comment (ramindu90) [reply modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx]: +1

File: modules/solutions/twitter-analytics/widgets/EmotionsAnalysis/src/resources/widgetConf.json [hunk 2]
```diff
@@ -7,6 +7,21 @@
...
         "subscriber"
       ]
+    },
+    "providerConfig": {
+      "configs": {
+        "type": "RDBMSBatchDataProvider",
+        "config": {
+          "datasourceName": "Twitter_Analytics",
+          "queryData": {
+            "query": "select AGG_TIMESTAMP as time, CAST(AGG_SUM_value AS DOUBLE)/AGG_COUNT as Average from TweetAggre_MINUTES where AGG_TIMESTAMP>{{filterCondition}} "
```
Comment (Niveathika92) [inline modules/solutions/twitter-analytics/widgets/EmotionsAnalysis/src/resources/widgetConf.json]: Shall we rename {{filterCondition}} as {{startTime}}

**Pull request merged**